packages/fuels-signers/src/wallet.rs

```diff
@@ -121,9 +118,16 @@ impl Wallet {
         }
     }
 
+    pub fn get_provider(&self) -> Result<&Provider, WalletError> {
+        self.provider.as_ref().ok_or(WalletError::NoProvider)
+    }
```

Comment:
I've been needing this a lot; requiring `Wallet`s to have a `Provider` at creation time was actually a major hassle!